lib/src/models/comment_impl.dart

```diff
@@ -139,25 +158,172 @@ class MoreComments extends RedditBase {
   }
 }
 
-class Comment extends CommentRef {
+/// A fully initialized class which represents a single Reddit comment.
+class Comment extends CommentRef
+    with
+        EditableMixin,
+        GildableMixin,
+        InboxToggleableMixin,
+        InboxableMixin,
+        RedditBaseInitializedMixin,
+        ReplyableMixin,
+        ReportableMixin,
+        SaveableMixin,
+        UserContentMixin,
+        VoteableMixin {
   // CommentModeration get mod; // TODO(bkonyi): implement
 
+  /// Has this [Comment] been approved.
+  bool get approved => data['approved'];
+
+  /// When this [Comment] was approved.
+  ///
+  /// Returns `null` if this [Comment] has not been approved.
+  DateTime get approvedAtUtc => (data['approved_at_utc'] == null)
+      ? null
+      : new DateTime.fromMillisecondsSinceEpoch(
+          data['approved_at_utc'].round() * 1000,
+          isUtc: true);
+
+  /// Which Redditor approved this [Comment].
+  ///
+  /// Returns `null` if this [Comment] has not been approved.
+  RedditorRef get approvedBy => (data['approved_by'] == null)
```

Comment:
Theoretical Question: (Disclaimer I don't really understand how in the world closures work, but I'm gonna ask a question that is probably going to sound stupid) If we use closures, does the new function created persist in the symbol table for the user session / or does it have to be generated again on every call to the getters?

In general though, I feel like we should have some sort of property extractor that is Synchronous in the base_impl.dart, because this null checking is getting out of hand (making getters which shouldn't have much logic, really hard to understand.

Reply:
See above comment for making a helper for these sort of getters.

Not sure what the context is for the closure comment. Mind elaborating here or offline?

Reply:
Added helper.